portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,11 +25,23 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private List<String> facts;
+
+  @Override
+  public void init() {
+      facts = new ArrayList<>();
```

Comment:
In Google coding style, we generally prefer to use immutable types (provided by Guava libraries).
See http://go/j-p/choosing-collections.

Using the ImmutableList type, your code could be simplified to a single call to ImmutableList.of("fact 1", "fact 2", "fact 3", "fact 4").